src/Microsoft.ApplicationInsights.WorkerService/ApplicationInsightsExtensions.cs

```diff
@@ -0,0 +1,206 @@
+        /// <param name="options">The action used to configure the options.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/>.
+        /// </returns>
+        public static IServiceCollection AddApplicationInsightsTelemetryWorkerService(
+            this IServiceCollection services,
+            Action<ApplicationInsightsServiceOptions> options)
+        {
+            services.AddApplicationInsightsTelemetryWorkerService();
+            services.Configure(options);
+            return services;
+        }
+
+        /// <summary>
+        /// Adds Application Insights services into service collection.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> instance.</param>
+        /// <param name="options">The options instance used to configure with.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/>.
+        /// </returns>
+        public static IServiceCollection AddApplicationInsightsTelemetryWorkerService(
+            this IServiceCollection services,
+            ApplicationInsightsServiceOptions options)
+        {
+            services.AddApplicationInsightsTelemetryWorkerService();
+            services.Configure((ApplicationInsightsServiceOptions o) =>
+            {
+                o.ApplicationVersion = options.ApplicationVersion;
+                o.DeveloperMode = options.DeveloperMode;
+                o.EnableAdaptiveSampling = options.EnableAdaptiveSampling;
+                o.EnableAuthenticationTrackingJavaScript = options.EnableAuthenticationTrackingJavaScript;
+                o.EnableDebugLogger = options.EnableDebugLogger;
+                o.EnableQuickPulseMetricStream = options.EnableQuickPulseMetricStream;
+                o.EndpointAddress = options.EndpointAddress;
+                o.InstrumentationKey = options.InstrumentationKey;
+                o.EnableHeartbeat = options.EnableHeartbeat;
+                o.AddAutoCollectedMetricExtractor = options.AddAutoCollectedMetricExtractor;
+            });
+            return services;
+        }
+
+        /// <summary>
+        /// Adds Application Insights services into service collection.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> instance.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/>.
+        /// </returns>
+        public static IServiceCollection AddApplicationInsightsTelemetryWorkerService(this IServiceCollection services)
+        {
+            try
+            {
+                if (!IsApplicationInsightsAdded(services))
+                {
+                    services
+                        .AddSingleton<ITelemetryInitializer, ApplicationInsights.AspNetCore.TelemetryInitializers.
+                            DomainNameRoleInstanceTelemetryInitializer>();
+                    services.AddSingleton<ITelemetryInitializer, ComponentVersionTelemetryInitializer>();
```

Comment:
there is a lot of common configuration between worker and aspnetcore: dependencies, quickpulse, few more moduls, channel, many initializers, appid provider. Can we have common code?

Otherwise, any change in this configuration will need to be duplicated and we'll for sure forget about tiny details like TryAddSignleton vs AddSingleton and others resulting in different behavior.